src/state.py

```diff
@@ -32,7 +34,7 @@ def add_sprite(self, sprite, position_x, position_y, scale):
         })
 
 
-    def add_option(self, label, next_state, required, acquired):
+    def add_option(self, label: str, next_state: 'State', required: dict, acquired: dict):
```

Comment:
`required` and `acquired` could be `Dict[str, dict]`; the nested dict cannot be specify because it depends on the key, history is str to bool, items is str to int.